File: Medium/2025-11-14-2536-Increment-Submatrices-by-One/solution.py
class Solution:
    def incrementSubmatrices(self, n: int, queries: List[List[int]]) -> List[List[int]]:
        mat = [[0] * n for _ in range(n)]
        
        for row1, col1, row2, col2 in queries:
            for i in range(row1, row2 + 1):
                for j in range(col1, col2 + 1):
                    mat[i][j] += 1
        
        return mat

File: Medium/2025-11-14-2536-Increment-Submatrices-by-One/solution.js
var incrementSubmatrices = function(n, queries) {
    const mat = Array.from({ length: n }, () => Array(n).fill(0));

    for (const [row1, col1, row2, col2] of queries) {
        for (let i = row1; i <= row2; i++) {
            for (let j = col1; j <= col2; j++) {
                mat[i][j]++;
            }
        }
    }

    return mat;
};